doc/getting_started/index.rst

```diff
@@ -174,104 +174,294 @@ Next, clone Zephyr and its :ref:`modules <modules>` into a new :ref:`west
 <west>` workspace named :file:`zephyrproject`. You'll also install Zephyr's
 additional Python dependencies.
 
+Python is used by the ``west`` meta-tool as well as by many scripts invoked by
+the build system. It is easy to run into package incompatibilities when
+installing dependencies at a system or user level. This situation can happen,
+for example, if working on multiple Zephyr versions at the same time. For this
+reason it is suggested to use `Python virtual environments`_.
+
+.. _Python virtual environments: https://docs.python.org/3/library/venv.html
+
 .. tabs::
 
    .. group-tab:: Ubuntu
 
-      #. Install west, and make sure :file:`~/.local/bin` is on your
-         :envvar:`PATH` :ref:`environment variable <env_vars>`:
+      .. tabs::
 
-         .. code-block:: bash
+         .. group-tab:: Install globally
 
-            pip3 install --user -U west
-            echo 'export PATH=~/.local/bin:"$PATH"' >> ~/.bashrc
-            source ~/.bashrc
+            #. Install west, and make sure :file:`~/.local/bin` is on your
+               :envvar:`PATH` :ref:`environment variable <env_vars>`:
 
-      #. Get the Zephyr source code:
+               .. code-block:: bash
 
-         .. code-block:: bash
+                  pip3 install --user -U west
+                  echo 'export PATH=~/.local/bin:"$PATH"' >> ~/.bashrc
+                  source ~/.bashrc
 
-            west init ~/zephyrproject
-            cd ~/zephyrproject
-            west update
+            #. Get the Zephyr source code:
 
-      #. Export a :ref:`Zephyr CMake package <cmake_pkg>`. This allows CMake to
-         automatically load boilerplate code required for building Zephyr
-         applications.
+               .. code-block:: bash
 
-         .. code-block:: console
+                  west init ~/zephyrproject
+                  cd ~/zephyrproject
+                  west update
 
-            west zephyr-export
+            #. Export a :ref:`Zephyr CMake package <cmake_pkg>`. This allows CMake to
+               automatically load boilerplate code required for building Zephyr
+               applications.
 
-      #. Zephyr's ``scripts/requirements.txt`` file declares additional Python
-         dependencies. Install them with ``pip3``.
+               .. code-block:: console
 
-         .. code-block:: bash
+                  west zephyr-export
+
+            #. Zephyr's ``scripts/requirements.txt`` file declares additional Python
+               dependencies. Install them with ``pip3``.
+
+               .. code-block:: bash
+
+                  pip3 install --user -r ~/zephyrproject/zephyr/scripts/requirements.txt
+
+         .. group-tab:: Install within virtual environment
+
+            #. Create a new virtual environment:
+
+               .. code-block:: bash
+
+                  python3 -m venv .venv
```

Comment:
Please specify the folder where you expect users to run this on

Reply:
(same for Windows/macOS)

Reply:
fixed, thanks